homeassistant/components/opower/sensor.py

```diff
@@ -171,11 +171,22 @@ async def async_setup_entry(
     coordinator: OpowerCoordinator = hass.data[DOMAIN][entry.entry_id]
     entities: list[OpowerSensor] = []
     forecasts = coordinator.data.values()
+
+    # Utility account id is not necessarily unique.  For backwards
+    # compatibility, when it is unique, use that in the id and name generation,
+    # but when it is not, use the uuid.
+    utility_account_ids = [
+        forecast.account.utility_account_id for forecast in forecasts
+    ]
+    duplicate_utility_account_ids = [
+        i for i in set(utility_account_ids) if utility_account_ids.count(i) > 1
+    ]
+
     for forecast in forecasts:
-        device_id = f"{coordinator.api.utility.subdomain()}_{forecast.account.utility_account_id}"
+        device_id = f"{coordinator.api.utility.subdomain()}_{forecast.account.uuid if forecast.account.utility_account_id in duplicate_utility_account_ids else forecast.account.utility_account_id}"
         device = DeviceInfo(
             identifiers={(DOMAIN, device_id)},
-            name=f"{forecast.account.meter_type.name} account {forecast.account.utility_account_id}",
+            name=f"{forecast.account.meter_type.name} account {forecast.account.uuid if forecast.account.utility_account_id in duplicate_utility_account_ids else forecast.account.utility_account_id}",
```

Comment:
The id is also used 3 lines above. Extract it to a local variable to avoid duplication.